### PR TITLE
[WIP] Support the standard JSON via compile() too

### PR DESCRIPTION
**Pull request by axic**
Part of #120 and https://github.com/ethereum/solidity/issues/3452

### CODE REVIEW

**Comment by axic** [comment]
Needs tests.

File: wrapper.js [hunk 1]
```diff
@@ -68,6 +68,28 @@ function setupMethods (soljson) {
...
+        input = JSON.parse(input);
+        if (typeof input['language'] === 'string') {
+          isStandardJSON = true;
+        }
+      } catch (e) {
+      }
+
+      if (isStandardJSON) {
+        // NOTE: takes second argument as "readCallback"
+        return JSON.parse(compileStandardWrapper(input, optimise));
```
Comment (chriseth) [inline wrapper.js]: `compileStandardWrapper` is only assigned in line 83 / 105

**Comment by axic** [comment]
> I am not convinced this is working.

Probably it isn't yet, it is more about discussing the API.

Eventually at 0.5.0 I'd like to replace `compile` to only do what `compileStandardWrapper` is doing. This meant to be a stepping stone to there.

**Comment by chriseth** [comment]
Yes, I think it is a good idea to make compile to also take json-io.

File: wrapper.js (hunk 1 above)
Comment (axic) [reply wrapper.js]: Javascript scoping... 😉

Comment (chriseth) [reply wrapper.js]: Ah, it's a callback!

**Comment by axic** [comment]
This needs tests.

File: README.md [hunk 2]
```diff
@@ -104,6 +104,19 @@ There is also a direct method, `compileStandard`, which is only present on recen
 
 Starting from version 0.4.20 a Semver compatible version number can be retrieved on every compiler release, including old ones, using the `semver()` method.
 
+#### From version 0.4.21
```
Comment (axic) [inline README.md]: Oh man, only if we'd have merged this at 0.4.21, it wouldn't be too big a breaking change anymore :)

**Pull request closed without merging**